registry/storage/driver/storagedriver.go

```diff
@@ -106,11 +106,11 @@ type FileWriter interface {
 }
 
 // PathRegexp is the regular expression which each file path must match. A
-// file path is absolute, beginning with a path components and may contain
-// more of them separated by slashes, where each path component is
+// file path is absolute, beginning with a slash or a path components and may
+// contain more of them separated by slashes, where each path component is
 // restricted to alphanumeric characters or a period, underscore, colon or
 // hyphen.
-var PathRegexp = regexp.MustCompile(`^([A-Za-z0-9._:-]+(/[A-Za-z0-9._:-]+)*)+$`)
+var PathRegexp = regexp.MustCompile(`^([A-Za-z0-9._:-]*(/[A-Za-z0-9._:-]+)*)+$`)
```

Comment:
i waned to use the actual path itself and the actual paths doesn't start with a `/` which is why the driver code is striping the starting `/` in the code https://github.com/docker/distribution/blob/master/registry/storage/driver/gcs/gcs.go#L868 https://github.com/docker/distribution/blob/master/registry/storage/driver/s3-aws/s3.go#L648
https://github.com/docker/distribution/blob/master/registry/storage/driver/swift/swift.go#L583

i wanted to support path that don't start with a `/` if i wanted to use this just as a storage wrapper and then if i wanted to use the whole repo as a private registry then during a docker push the paths coming to the storage driver are starting with `/` which is the reason i want to support path that start with `/` even though ultimately the storage driver is going to strip it off.
Basically my previous PR and this PR aims to change the regex such that it validates/matches both the paths which start or doesn't start with a `/`.